src/DropDownMenu/DropDownMenu.jsx

```diff
@@ -0,0 +1,379 @@
+        </ClearFix>
+        <Popover
+          anchorOrigin={{horizontal: 'left', vertical: 'top'}}
+          anchorEl={anchorEl}
+          style={popoverStyle}
+          animation={PopoverAnimationFromTop}
+          open={open}
+          onRequestClose={this._onMenuRequestClose} >
+          <Menu
+            maxHeight={maxHeight}
+            desktop={true}
+            value={value}
+            style={menuStyle}
+            >
+            {menuItemElements}
+          </Menu>
+        </Popover>
+      </div>
+    );
+  },
+
+  _testDeprecations() {
+    warning(!this.props.hasOwnProperty('displayMember'),
+      'displayMember will be removed in favor of composability. refer to the documentation for more information');
+
+    warning(!this.props.hasOwnProperty('labelMember'),
+      'labelMember will be removed in favor of composability. refer to the documentation for more information');
+
+    warning(!this.props.hasOwnProperty('labelStyle'),
+      'labelStyle will be removed with labelMember.');
+
+    warning(!this.props.hasOwnProperty('menuItems'),
+      'menuItems will be removed in favor of composability. refer to the documentation for more information');
+
+    warning(!this.props.hasOwnProperty('selectedIndex'),
+      'selectedIndex will be removed. use value instead to control the component.');
+
+    warning(!this.props.hasOwnProperty('valueLink'),
+      'valueLink will be removed. use value and onChange');
+
+    warning(!this.props.hasOwnProperty('valueMember'),
+      'valueMember will be removed in favor of composability. refer to the documentation for more information');
+  },
+
+  _setWidth() {
+    const el = ReactDOM.findDOMNode(this);
+    if (!this.props.style || !this.props.style.hasOwnProperty('width')) {
+      el.style.width = 'auto';
+    }
+  },
+
+  _setSelectedIndex(index) {
+    if (process.env.NODE_ENV !== 'production') {
+      warning(index >= 0, 'Cannot set selectedIndex to a negative index.');
+    }
+
+    this.setState({selectedIndex: (index >= 0) ? index : 0});
+  },
+
+  _onControlTouchTap(event) {
+    event.preventDefault();
+    if (!this.props.disabled) {
+      this.setState({
+        open: !this.state.open,
+        anchorEl: ReactDOM.findDOMNode(this),
+      });
+    }
+  },
+
+  _onMenuItemTouchTap(key, payload, e) {
+    const {
+      onChange,
+      menuItems,
+      value,
+      valueLink,
+      valueMember,
+    } = this.props;
+
+    if (menuItems && (this.state.selectedIndex !== key || e.target.value !== value)) {
+      const selectedItem = menuItems[key];
+      if (selectedItem) {
+        e.target.value = selectedItem[valueMember];
+      }
+
+      if (valueLink) {
+        valueLink.requestChange(e.target.value);
+      }
+      else if (onChange) {
+        onChange(e, key, payload);
+      }
+      this._onMenuRequestClose();
+    } else if (onChange) {
+      if (valueLink) {
+        valueLink.requestChange(e.target.value);
```

Comment:
Regarding this block of code:

```js
if (valueLink) {
  valueLink.requestChange(e.target.value)
} else if (onChange) {
  onChange(e, key, payload);
}
```

We could probably just put it at the end of the `_onMenuItemTouchTap` and remove the two instances of the block in the if/else